tests/components/bthome/test_sensor.py

```diff
@@ -0,0 +1,517 @@
+"""Test the BThome sensors."""
+
+from unittest.mock import patch
+
+from homeassistant.components.bluetooth import BluetoothChange
+from homeassistant.components.bthome.const import DOMAIN
+from homeassistant.components.sensor import ATTR_STATE_CLASS
+from homeassistant.const import ATTR_FRIENDLY_NAME, ATTR_UNIT_OF_MEASUREMENT
+
+from . import TEMP_HUMI_SERVICE_INFO, make_advertisement, make_encrypted_advertisement
+
+from tests.common import MockConfigEntry
+
+
+async def test_sensors(hass):
```

Comment:
I'd recommend using pytest.mark.parametrize to parametrize these tests as they are very similar.

https://docs.pytest.org/en/stable/how-to/parametrize.html

Reply:
Done.